src/coinbase/types.ts

```diff
@@ -812,14 +812,87 @@ export enum FundOperationStatus {
 }
 
 /**
- * The Wallet Data type definition.
- * The data required to recreate a Wallet.
+ * Interface representing wallet data, with support for both camelCase and snake_case
+ * property names for compatibility with older versions of the Python SDK.
  */
-export type WalletData = {
-  walletId: string;
+export interface WalletData {
+  /**
+   * The CDP wallet ID in either camelCase or snake_case format, but not both.
+   */
+  walletId?: string;
+  wallet_id?: string;
```

Comment:
One option to try to enforce mutual exclusivity is to use `never`:
```ts
type WalletData = {
    walletId: string;
    wallet_id?: never;
} | {
    walletId?: never;
    wallet_id: string;
}

const walletData: WalletData = {
    walletId: '123',
    wallet_id: '123'
}
```

Now TypeScript will error with a message:
```
Type '{ walletId: string; wallet_id: string; }' is not assignable to type 'WalletData'.
  Types of property 'wallet_id' are incompatible.
    Type 'string' is not assignable to type 'undefined'
```

Admittedly, the error message could be clearer, but it's difficult to customize the message without reaching for a type validation library like Zod, which might be overkill here.